src/main/java/hudson/plugins/ansicolor/AnsiColorConsoleLogFilter.java

```diff
@@ -20,10 +24,40 @@
     private static final Logger LOG = Logger.getLogger(AnsiColorConsoleLogFilter.class.getName());
 
     private AnsiColorMap colorMap;
+    private final Map<String, byte[]> notes;
 
     public AnsiColorConsoleLogFilter(AnsiColorMap colorMap) {
         super();
         this.colorMap = colorMap;
+        this.notes = new HashMap<>();
+        // some cases of AnsiHtmlOutputStream.setForegroundColor:
+        for (AnsiColorMap.Color color : AnsiColorMap.Color.values()) {
+            pregenerateNote(new AnsiAttributeElement(AnsiAttributeElement.AnsiAttrType.FG, "span", "style=\"color: " + colorMap.getNormal(color.ordinal()) + ";\""));
+        }
+        // TODO other cases, and other methods
+        LOG.log(Level.FINE, "Notes pregenerated for {0}", notes.keySet());
+    }
+    
+    private void pregenerateNote(AnsiAttributeElement element) {
+        element.emitOpen(html -> pregenerateNote(html));
+        element.emitClose(html -> pregenerateNote(html));
+    }
+    
+    private void pregenerateNote(String html) {
+        if (!notes.containsKey(html)) {
+            JenkinsJVM.checkJenkinsJVM();
```

Comment:
See https://github.com/jenkinsci/ant-plugin/pull/32 for a simpler patch that illustrates the idiom.